corehq/apps/app_manager/tasks.py

```diff
@@ -23,26 +23,32 @@ def create_user_cases(domain_name):
 
 
 @serial_task('{app_id}-{version}', max_retries=0, timeout=60 * 60)
-def make_async_build_v2(app_id, domain, version, allow_prune=True, comment=None):
+def make_async_build_v2(app_id, domain, version, username, allow_prune=True, comment=None):
     app = get_app(domain, app_id)
-    return make_async_build(app, allow_prune=allow_prune, comment=comment)
-
-
-@serial_task('{app._id}-{app.version}', max_retries=0, timeout=60 * 60)
-def make_async_build(app, allow_prune=True, comment=None):
-    previous_version = app.get_previous_version()
-    if previous_version and previous_version.version == app.version:
-        return
     errors = app.validate_app()
     if not errors:
         if allow_prune and comment is None:
             comment = _('Auto-generated by a phone update. Will expire after next build if not marked released.')
+        if username:
+            comment += _('Generated by user {}').format(username)
         copy = app.make_build(comment=comment)
         copy.is_auto_generated = allow_prune
         copy.save(increment_version=False)
         return copy
 
 
+def make_build(app, username, allow_prune=True, comment=None, async=True):
```

Comment:
This is used in two different ways:
* Automatically kick off a build of the latest saved version - `allow_prune=True, comment=None, async=True`
* Synchronously build a template app - `allow_prune=False, comment="...", async=False`

These two use-cases seem different enough to warrant their own functions.  That'd likely simplify the situation a bit, since there would be two sets of args, none of which is optional, and only one set needs to be a task.

Reply:
https://github.com/dimagi/commcare-hq/pull/24860/commits/d78c68d1651927faa00cce4bca0bda530b919a4d

Reply:
https://github.com/dimagi/commcare-hq/pull/24860/commits/5fe187929ca7c15c8707d1501f0a1e32f1323d54